### ISSUE TITLE
Climate

### ISSUE THREAD

**Issue by avworldme** [laurent-martin/ets-to-homeassistant#11]
Can this be used to import climate device settings from an ETS file ?

**Comment by laurent-martin** [comment]
Mmm.. no, currently only switches

Nevertheless, if you feel it, you can develop a "custom lambda" which will receive all group addresses from your project, and you have an opportunity to create home assistant objects from there.

**Comment by avworldme** [comment]
Thank you for the reply.  I will try it although I don't know much ruby.  Any clues or instructions?

**Comment by TheFelix93** [comment]
I figured out how I can adapt the script to my needs. For me it was easier change the main script instead of doing all in the custom-lambda.

I added my code into map_ets_datapoint_to_ha_type() +  some changes in the lambda script "laurent_specific.rb".
I implemented TW Strips, RGBW Strips, binary sensors and other sensors like temp or humidity, cover. Climate is still on my TODO.

To identify which GA is for which function and which datapoints and HA-Attributes I use:
- Function-Type from ETS (in ets5 there are only 6 types, I use the most matching for all functions and if not possible I use the custom type.
- Inside the functions I use the GA-Name and the middle part of the GA to identify the exact HA-attribute (depends on your GA-Structure inside ETS)

At the moment all I did is spaghetti code, but when I am done with my home installation I can refactor it a little bit and make a pull-request.

On the other hand I am bit mad, that all this was not done by the author of that repo already. The code that initially reads the KNX-project-file is actually the hardest part. I could imagine, that its not worth it because the HA-Team might already be working hard on this, but we never know :)

Nevertheless I am really grateful that Martin created this script. With this I was able to adapt it to my ETS project. I already have 98 entities generated, which would be really painful if it was manual work...

**Comment by Superwallah** [comment]
@TheFelix93 You did not commit your changes to your github, do you? Would be nice to see what you have achieved.

**Comment by TheFelix93** [comment]
no, unfortunately still not. I cloned the repo shortly before Martin refactored the whole script. Now I have to do a bit more to integrate it.
I am not a ruby expert and also long time out of the coding business... I need more time than expected, but I will definitely do it.

**Comment by TheFelix93** [comment]
> @TheFelix93 You did not commit your changes to your github, do you? Would be nice to see what you have achieved.

I finally made it. See my fork: [https://github.com/TheFelix93/ets-to-homeassistant/tree/master](https://github.com/TheFelix93/ets-to-homeassistant/tree/master)

**Comment by TheFelix93** [comment]
> Can this be used to import climate device settings from an ETS file ?

See my fork supports climate now.
https://github.com/TheFelix93/ets-to-homeassistant
